### PR TITLE
Tasks/uber barber #29 #28 #25 user validation

### LINKED ISSUE #25: Password at min. 6 letters + not clickable confirm button without if passwords does not match - USER REGISTRATION

### LINKED ISSUE #28: Not clickable button if there is no username AND password - LOGGING

### LINKED ISSUE #29: Username should has only letters, numbers and dots maybe? also min. 6 characters - USER REGISTRATION

### PR DESCRIPTION
**Pull request by MSkrzypinski**
### Description
Added user validation.

User registration:
- Username can only contain letters, numbers, dots
- Username must be at least 6 characters
- Password must be at least 6 letters
- Disabled user registration confirm button if passwords doesn't match
- Added email validation

User login:

- Disabled login button if username and password are empty

#### Checklist for Developer
- [x] WPF issue
- [x] .cs issue 
- [ ] MySql issue
- [ ] Document issue
- [ ] Test issue
- [x] Add reviewers to pull-request
- [x] Assign yourself to pull-request
- [x] Add labels to pull-request
- [x] Link an issue to from repository

Please note that all items which are listed above are relevant to merge the pull-request!

If the checkboxes are empty pull-request will be closed!

### CODE REVIEW

**Comment by Borzygniew** [comment]
Complete checklist for developer by clicking on checkbox or adding x in brackets like this -> [x]
Also assign yourself and link an issue. Under Reviewers -> Assignees, Under Labels -> Development

File: UberBarber/UberBarber/database/DatabaseQueries.cs [hunk 1]
```diff
@@ -69,14 +70,35 @@ public string User_validation(string username, string password, string confirm_p
...
+            // username validation
+            else if (!Regex.IsMatch(username, usernamePattern))
+            {
+                message = "Username can only contain letters, numbers, dots and have at least 6 characters";
+                return message;
+            }
+            //password validation
+            else if (!Regex.IsMatch(password, passwordPattern))
+            {
+                message = "Password must contain at least 6 letters";
```
Comment (Borzygniew) [inline UberBarber/UberBarber/database/DatabaseQueries.cs]: letters only? why not characters?

File: UberBarber/UberBarber/database/DatabaseQueries.cs [hunk 2]
```diff
@@ -69,14 +70,35 @@ public string User_validation(string username, string password, string confirm_p
             // Returns proper message when the validation is correct or not.
         {
             string message = "Something went wrong :(";
+            string usernamePattern = "^[a-zA-Z0-9.]{6,}$";
+            string passwordPattern = "^[a-zA-Z]{6,}$";
```
Comment (Borzygniew) [inline UberBarber/UberBarber/database/DatabaseQueries.cs]: is this a pattern for only letters in password? add numbers here

**Review by Borzygniew — CHANGES_REQUESTED** [review]
Good pull request

File: UberBarber/UberBarber/View/UserRegistration.xaml [hunk 3]
```diff
@@ -219,8 +221,8 @@
                                         AlignmentX="Left"/>
                         </TextBox.Background>
                     </TextBox>
-                    <!-- Label with error info-->
-                    <Label x:Name="LabelInfoUserReg" FontFamily="Montserrat" Foreground="Red" FontSize="12" HorizontalAlignment="Center"/>
+                    <!-- TextBlock with error info-->
+                    <TextBlock x:Name="TextBlockInfoUserReg" FontFamily="Montserrat" Margin="0,5,0,-5" Foreground="Red" FontSize="12" HorizontalAlignment="Center" TextWrapping="Wrap"/>
```
Comment (goflyy) [inline UberBarber/UberBarber/View/UserRegistration.xaml]: Keep the code "clean". X:name, FontFamily, ForeGround... write one after the other

File: UberBarber/UberBarber/database/DatabaseQueries.cs (hunk 2 above)
Comment (MSkrzypinski) [reply UberBarber/UberBarber/database/DatabaseQueries.cs]: Numbers added to the pattern.

File: UberBarber/UberBarber/database/DatabaseQueries.cs (hunk 1 above)
Comment (MSkrzypinski) [reply UberBarber/UberBarber/database/DatabaseQueries.cs]: Corrected

File: UberBarber/UberBarber/View/UserRegistration.xaml (hunk 3 above)
Comment (MSkrzypinski) [reply UberBarber/UberBarber/View/UserRegistration.xaml]: Corrected

**Review by Borzygniew — APPROVED** [review]
Zrzut ekranu 2023-01-11 213744

**Review by goflyy — APPROVED** [review]

**Pull request merged**